src/components/search/GlobalSearchField.js

```diff
@@ -130,22 +134,52 @@ const SearchOption = React.forwardRef((props, ref) => {
                 }
                 primaryTypographyProps={{
                     variant: "body1",
-                    color: "primary",
+                    color: theme.palette.info.primary,
                 }}
                 secondary={
                     <Highlighter search={searchTerm}>{secondary}</Highlighter>
                 }
                 secondaryTypographyProps={{
                     variant: "caption",
                     wrap: "true",
+                    color: theme.palette.info.primary,
                 }}
             />
         </ListItem>
     );
 });
 
+function ViewAllOption(props) {
+    const { id, searchTerm, filter, prompt, selectedTab } = props;
+    const href = `${NavigationConstants.SEARCH}?searchTerm=${searchTerm}&filter=${filter}&selectedTab=${selectedTab}`;
+    const as = `${NavigationConstants.SEARCH}?searchTerm=${searchTerm}&filter=${filter}&selectedTab=${selectedTab}`;
+    return (
+        <Link href={href} as={as} passHref>
+            <SearchOption
+                primary={searchTerm}
+                secondary={prompt}
+                icon={<PageviewIcon color="primary" />}
+                id={id}
+            />
+        </Link>
+    );
+}
+
 function DataSearchOption(props) {
-    const { selectedOption, searchTerm } = props;
+    const { baseId, filter, selectedOption, searchTerm } = props;
+    const { t: i18NSearch } = useTranslation("search");
+
+    if (selectedOption?.id === searchConstants.VIEW_ALL_ID) {
+        return (
+            <ViewAllOption
+                searchTerm={searchTerm}
+                filter={filter}
+                prompt={i18NSearch("viewAllDataResults", { searchTerm })}
+                selectedTab={SEARCH_RESULTS_TABS.data}
+                id={build(baseId, ids.VIEW_ALL)}
+            />
+        );
```

Comment:
When `View All` option is selected, users will be taken to detailed search results page and corresponding tab is selected.